src/vm.rs

```diff
@@ -707,14 +707,9 @@ impl<T: GuestStagePageTable> Vm<T, VmStateFinalized> {
             }
         }
 
-        fn read_firmware_counter(counter_index: u64) -> EcallResult<u64> {
-            use sbi::api::pmu;
-            let pmu_info = PmuInfo::get();
-            if pmu_info.get_counter_info(counter_index).is_some() {
-                match pmu::read_firmware_counter(counter_index) {
-                    Ok(n) => Ok(n),
-                    Err(e) => Err(EcallError::Sbi(e)),
-                }
+        fn read_firmware_counter(counter_index: u64, active_cpu: &VmCpu) -> EcallResult<u64> {
```

Comment:
General comment with respect to firmware counters: should the TSM even expose them at all? All the firmware counters are for actions that are delegated to firmware (e.g. remote fences) or for alignment/instruction traps that get silently fixed up by firmware. The TSM isn't doing any of that for TVMs so, theoretically, these counters should be no-ops.

Reply:
No sure...but think we'll need to return 0s or something like that, right? @atishp04 Do you have any thoughts on the matter?

Reply:
Just don't return any firmware counters in the counter_info call. That is a sufficient hint for the lower privilege mode to know that there are no firmware counters available. In that case, read_firmware_counter should return -EINVAL as no firmware counter available.

config_match should return SBI_ERR_NOT_SUPPORTED for any firmware events as well.

The current set of firmware counters are pretty much useless for TSM. In KVM, I have only implemented few only.